comms/src/connection/mod.rs

```diff
@@ -20,10 +20,12 @@
 //  WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
 //  USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
+#[macro_use]
```

Comment:
dont think this is required for 2018 edition

Reply:
You still need macro_use to use macros from the crates own modules. https://doc.rust-lang.org/edition-guide/rust-2018/macros/macro-changes.html - looks like this is going to change eventually, but right now it's needed